gym/spaces/utils.py

```diff
@@ -118,6 +131,24 @@ def _flatten_dict(space, x) -> np.ndarray:
     return np.concatenate([flatten(s, x[key]) for key, s in space.spaces.items()])
 
 
+@flatten.register(Graph)
+def _flatten_graph(space, x) -> np.ndarray:
+    def _graph_unflatten(space, x):
+        ret = None
+        if space is not None and x is not None:
+            if isinstance(space, Box):
+                ret = x.reshape(x.shape[0], -1)
```

Comment:
Can we call `flatten(x, space)` for both Box and Discrete here? Just in case the specifications of Box and Discrete change